exploration-module/src/main/java/io/edpn/backend/exploration/application/service/ReceiveJournalScanService.java

```diff
@@ -11,45 +11,46 @@
 import io.edpn.backend.exploration.application.port.outgoing.system.SaveOrUpdateSystemPort;
 import io.edpn.backend.messageprocessorlib.application.dto.eddn.journal.ScanMessage;
 import io.edpn.backend.util.IdGenerator;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.transaction.annotation.Transactional;
+
 import java.util.ArrayList;
 import java.util.Map;
 import java.util.Optional;
 import java.util.UUID;
 import java.util.stream.Collectors;
-import lombok.RequiredArgsConstructor;
-import lombok.extern.slf4j.Slf4j;
-import org.springframework.transaction.annotation.Transactional;
 
 @RequiredArgsConstructor
 @Slf4j
 public class ReceiveJournalScanService implements ReceiveKafkaMessageUseCase<ScanMessage.V1> {
-    
+
     private final IdGenerator idGenerator;
     private final SaveOrUpdateBodyPort saveOrUpdateBodyPort;
     private final SaveOrUpdateRingPort saveOrUpdateRingPort;
     private final SaveOrUpdateStarPort saveOrUpdateStarPort;
     private final SaveOrUpdateSystemPort saveOrUpdateSystemPort;
-    
+
     @Transactional
     @Override
     public void receive(ScanMessage.V1 message) {
         long start = java.lang.System.nanoTime();
         log.debug("DefaultReceiveJournalScanMessageUseCase.receive -> BodyMessage: {}", message);
-        
+
         ScanMessage.V1.Payload payload = message.message();
-        
+
         createOrUpdateFromPayload(payload);
-        
+
         log.trace("DefaultReceiveJournalScanMessageUseCase.receive -> took {} nanosecond", java.lang.System.nanoTime() - start);
         log.info("DefaultReceiveJournalScanMessageUseCase.receive -> the message has been processed");
     }
-    
-    
+
+
     private void createOrUpdateFromPayload(ScanMessage.V1.Payload payload) {
-        
+
```

Comment:
I think the if/else choice of star/body suggests, that perhaps this would be a case for polymorphism...

Reply:
this is comming from EDDN, so while there could be some way to further define superclasses etc,
it would tightly couple the different types of scan message, so I would advise against that. But feel free to give it a go :D

https://github.com/ed-pilots-network/backend-messageprocessor-lib/blob/development/src/main/java/io/edpn/backend/messageprocessorlib/application/dto/eddn/journal/ScanMessage.java